packages/editor-ui/src/components/BreakpointsObserver.vue

```diff
@@ -16,85 +19,72 @@ import { BREAKPOINT_SM, BREAKPOINT_MD, BREAKPOINT_LG, BREAKPOINT_XL } from '@/co
  * lg >= 1200
  * xl >= 1920
  */
+interface Props {
+	valueXS?: number;
+	valueXL?: number;
+	valueLG?: number;
+	valueMD?: number;
+	valueSM?: number;
+	valueDefault?: number;
+}
 
-import { useUIStore } from '@/stores/ui.store';
-import { getBannerRowHeight } from '@/utils/htmlUtils';
-import { useDebounce } from '@/composables/useDebounce';
-
-export default defineComponent({
-	name: 'BreakpointsObserver',
-	props: ['valueXS', 'valueXL', 'valueLG', 'valueMD', 'valueSM', 'valueDefault'],
-	setup() {
-		const { callDebounced } = useDebounce();
-		return { callDebounced };
-	},
-	data() {
-		return {
-			width: window.innerWidth,
-		};
-	},
-	computed: {
-		bp(): string {
-			if (this.width < BREAKPOINT_SM) {
-				return 'XS';
-			}
+const props = defineProps<Props>();
 
-			if (this.width >= BREAKPOINT_XL) {
-				return 'XL';
-			}
+const { callDebounced } = useDebounce();
+const width = ref(window.innerWidth);
 
-			if (this.width >= BREAKPOINT_LG) {
-				return 'LG';
-			}
-
-			if (this.width >= BREAKPOINT_MD) {
-				return 'MD';
-			}
-
-			return 'SM';
-		},
-
-		value(): number | undefined {
-			if (this.valueXS !== undefined && this.width < BREAKPOINT_SM) {
-				return this.valueXS;
-			}
+const bp = computed(() => {
+	if (width.value < BREAKPOINT_SM) {
+		return 'XS';
+	}
+	if (width.value >= BREAKPOINT_XL) {
+		return 'XL';
+	}
+	if (width.value >= BREAKPOINT_LG) {
+		return 'LG';
+	}
+	if (width.value >= BREAKPOINT_MD) {
+		return 'MD';
+	}
+	return 'SM';
+});
 
-			if (this.valueXL !== undefined && this.width >= BREAKPOINT_XL) {
-				return this.valueXL;
-			}
+const value = computed(() => {
+	if (props.valueXS && width.value < BREAKPOINT_SM) {
+		return props.valueXS;
+	}
+	if (props.valueXL && width.value >= BREAKPOINT_XL) {
+		return props.valueXL;
+	}
+	if (props.valueLG && width.value >= BREAKPOINT_LG) {
+		return props.valueLG;
+	}
+	if (props.valueMD && width.value >= BREAKPOINT_MD) {
+		return props.valueMD;
+	}
+	if (props.valueSM) {
+		return props.valueSM;
+	}
+	return props.valueDefault;
+});
 
-			if (this.valueLG !== undefined && this.width >= BREAKPOINT_LG) {
-				return this.valueLG;
-			}
+const onResize = () => {
+	void callDebounced(onResizeEnd, { debounceTime: 50 });
+};
 
-			if (this.valueMD !== undefined && this.width >= BREAKPOINT_MD) {
-				return this.valueMD;
-			}
+const onResizeEnd = async () => {
+	width.value = window.innerWidth;
+	await nextTick();
 
-			if (this.valueSM !== undefined) {
-				return this.valueSM;
-			}
+	const bannerHeight = await getBannerRowHeight();
+	useUIStore().updateBannersHeight(bannerHeight);
```

Comment:
Maybe we could have the `uiStore` initialized at root level to no longer make this function call every time we resize?

Reply:
Fair point, updated